framework/wazuh/core/utils.py

```diff
@@ -968,6 +969,23 @@ def check_section(agents_regex, split_section):
         check_section(auth_section, split_section='</auth>')
 
 
+def check_indexer(data: str):
+    """Check if modifying the indexer configuration is allowed.
+
+    Parameters
+    ----------
+    data : str
+        Configuration file content.
+    """
+    blocked_configurations = configuration.api_conf['upload_configuration']
+
+    if not blocked_configurations['indexer']['allow']:
+        with contextlib.suppress(IndexError):
+            indexer_section = re.compile(r"<indexer>(.*)</indexer>", flags=re.MULTILINE | re.DOTALL)
+            if len(indexer_section.findall(data)) > 0:
+                raise WazuhError(1130)
```

Comment:
In this case, the chosen approach should be more similar to `check_wazuh_limits_unchanged(...)` (https://github.com/wazuh/wazuh/pull/16541) than to `check_agents_allow_higher_versions(...)` (https://github.com/wazuh/wazuh/issues/18139). 

Notice that, by default, all `ossec.conf` files will contain the new `<indexer>` block. It means that when a user tries to edit the configuration through the dashboard, it will get its current configuration (`GET /manager/configuration`), modify it, and upload it as a new one (`PUT /manager/configuration`). 

Even if users change something different than the indexer section (let's say `<logall>`), the uploaded configuration will still contain all the `ossec.conf`, including said `indexer` block. This would lead to the `Indexer section modification attempt detected` exception. 

Therefore, uploading `ossec.conf` files containing the `<indexer>` section should be allowed as long as it remains unchanged (according to the `api.yaml`)